### ISSUE TITLE
[Training Loss Oscillation] Janus_VLN (Qwen2.5-VL + VGGT) training loss keeps oscillating on 8×B200 GPUs — seeking official suggestions

### ISSUE THREAD

**Issue by leopard-chen** [MIV-XJTU/JanusVLN#17]
1. Summary
First of all, thank you for the excellent work on Janus and Qwen2.5-VL — the framework is impressive and very helpful for our research.
I am training Janus_VLN (Qwen2.5-VL + VGGT) on 8×B200 GPUs, but the training loss shows strong oscillation and does not converge. I would like to seek your suggestions.

2. Environment

GPUs: 8×B200（CUDA_VISIBLE_DEVICES=1,2,3,4,5,6,7）

Precision: bf16

DeepSpeed: ZeRO-2

Model: Qwen2.5-VL + VGGT-1B

Dataset: train_r2r_rxr

3. Training Command
torchrun --nproc_per_node=7 \
  --master_addr=$MASTER_ADDR \
  --master_port=$MASTER_PORT \
  src/qwen_vl/train/train_qwen.py \
  --model_name_or_path $MODEL_PATH \
  --vggt_model_path $VGGT_MODEL_PATH \
  --tune_mm_llm True --tune_mm_vision False --tune_mm_mlp True \
  --dataset_use train_r2r_rxr \
  --output_dir ./JanusVLN_Base_new --cache_dir ./cache \
  --bf16 --per_device_train_batch_size 1 \
  --gradient_accumulation_steps 12 \
  --learning_rate 1e-6 \
  --mm_projector_lr 5e-6 \
  --vision_tower_lr 1e-6 \
  --model_max_length 163840 \
  --max_pixels $((576*28*28)) --min_pixels $((16*28*28)) \
  --video_max_frames 8 --video_min_frames 4 \
  --video_max_frame_pixels $((1664*28*28)) \
  --video_min_frame_pixels $((256*28*28)) \
  --num_train_epochs 1 \
  --warmup_ratio 0.03 \
  --lr_scheduler_type cosine \
  --deepspeed scripts/zero2.json \
  --gradient_checkpointing \
  --group_by_modality_length true \
  --seed 42

**Comment by missTL** [comment]
We usually use 64 GPUs or more for training. If you only have 8 GPUs, it is recommended to increase the `per_device_train_batch_size` and `gradient_accumulation_steps`.

**Comment by leopard-chen** [comment]
Hi, thanks for your suggestion about increasing per_device_train_batch_size and gradient_accumulation_steps.
I have one follow-up question:
When increasing these parameters, should the learning rate also be adjusted, or should I keep the same LR as long as the global batch size remains unchanged (e.g., still 512)?
In other words:
If I keep global_batch = per_device_batch_size × world_size × gradient_accumulation_steps = 512, is it correct that the learning rate does not need to change?
If the global batch size is decreased (e.g., <512), would you recommend scaling the learning rate, or keeping a smaller LR to avoid oscillation?
Thanks for your help!

**Comment by missTL** [comment]
If the total batch size is the same, the learning rate should also be the same. Otherwise, the learning rate should be adjusted accordingly. However, regarding the optimal batch size and learning rate, we haven't made many adjustments for verification, and they may not be the best.
